packages/minifier-css/minifier.js

```diff
@@ -116,96 +213,77 @@ var rewriteRules = function (rules, mergedCssPath) {
+    var cssUrlRegex = /url\s*\(\s*(['"]?)(.+?)\1\s*\)/gi;
+    let parts;
+    while (parts = cssUrlRegex.exec(value)) {
+      const oldCssUrl = parts[0];
+      const quote = parts[1];
+      const resource = url.parse(parts[2]);
 
-        // We used to finish the rewriting process at the absolute path step
-        // above. But it didn't work in case the Meteor application was deployed
-        // under a sub-path (eg `ROOT_URL=http://localhost:3000/myapp meteor`)
-        // in which case the resources linked in the merged CSS file would miss
-        // the `myapp/` prefix. Since this path prefix is only known at launch
-        // time (rather than build time) we can't use absolute paths to link
-        // resources in the generated CSS.
-        //
-        // Instead we transform absolute paths to make them relative to the
-        // merged CSS, leaving to the browser the responsibility to calculate
-        // the final resource links (by adding the application deployment
-        // prefix, here `myapp/`, if applicable).
-        relativeToMergedCss = pathRelative(mergedCssPath, absolutePath);
-        newCssUrl = "url(" + quote + relativeToMergedCss + quote + ")";
-        value = value.replace(oldCssUrl, newCssUrl);
+      // We don't rewrite URLs starting with a protocol definition such as
+      // http, https, or data, or those with network-path references
+      // i.e. //img.domain.com/cat.gif
+      if (resource.protocol !== null
+          || resource.href.startsWith('//')
+          || resource.href.startsWith('#')) {
+        continue;
       }
 
-      declaration.value = value;
-    });
+      // Rewrite relative paths (that refers to the internal application tree)
+      // to absolute paths (addressable from the public build).
+      let absolutePath = isRelative(resource.path)
+        ? pathJoin(basePath, resource.path)
+        : resource.path;
+
+      if (resource.hash) {
+        absolutePath += resource.hash;
+      }
+
+      // We used to finish the rewriting process at the absolute path step
+      // above. But it didn't work in case the Meteor application was deployed
+      // under a sub-path (eg `ROOT_URL=http://localhost:3000/myapp meteor`)
+      // in which case the resources linked in the merged CSS file would miss
+      // the `myapp/` prefix. Since this path prefix is only known at launch
+      // time (rather than build time) we can't use absolute paths to link
+      // resources in the generated CSS.
+      //
+      // Instead we transform absolute paths to make them relative to the
+      // merged CSS, leaving to the browser the responsibility to calculate
+      // the final resource links (by adding the application deployment
+      // prefix, here `myapp/`, if applicable).
+      const relativeToMergedCss = pathRelative(mergedCssPath, absolutePath);
+      const newCssUrl = 'url(' + quote + relativeToMergedCss + quote + ')';
```

Comment:
Unless we really need to preserve the `quote` character, this can just be
```
'url(' + JSON.stringify(relativeToMergedCss) + ')'
```

Reply:
Right, I don't think we need to preserve it, but I'll check (that was old code). If not, I'll update - thanks!

Reply:
I looked into replacing this, but it looks like the [current tests](https://github.com/meteor/meteor/blob/70ffce52fc36e91e188cc1d10a67a63f9969f5dd/packages/minifier-css/urlrewriting-tests.js) go to lengths to make sure the quote is preserved exactly as is. So either no quote, exactly a single quote, or exactly a double quote. There are tests verifying this fairly extensively. I've tested the results of `JSON.stringify` in practice, and everything seems to be okay, but I'm a bit concerned that the tests were added for a specific reason (maybe browser compatibility issues, special filenames / paths, etc.?). Given this, I'm wondering if we should leave this as is?